exporter/otlphttpexporter/otlp.go

```diff
@@ -85,6 +86,24 @@ func newExporter(cfg config.Exporter, logger *zap.Logger) (*exporterImp, error)
 	}, nil
 }
 
+// start overrides defaults base exporters start (no op) to attach auth transport to the http client.
+// This is the only place we get hold of Extensions which are required to construct auth round tripper.
```

Comment:
This is a shame that we're unable to modify this at a higher place, so that exporters do not need to be modified for auth extension support. It's not terrible, but I wish it could be implemented in a reusable place.

@jpkrohling @bogdandrutu @tigrannajaryan - Any thoughts on this?

Reply:
@pavankrish123, you could probably move the creation of the HTTP client to the start function, and pass the extensions to the ToClient() function, similar to what we do in the server part.

https://github.com/open-telemetry/opentelemetry-collector/blob/51281a7192564f4ce0c7b93978a1511bcf3ead7e/config/configgrpc/configgrpc.go#L244

The exporters would then not need to deal with the auth at all.

Reply:
@jpkrohling - This would be the ideal solution